### PR TITLE
navigate to subgoal with one click and moved goalItem summary

### LINKED ISSUE #1924: Navigate with one click
**Is your feature request related to a problem? Please describe.**
It's very annoying when you have to click twice to navigate down a goal hierarchy.

**Describe the solution you'd like**
1. Move the current subtext to below the title of the modal that appears when you click on the circle.
2. Also show this subtext underneath the title of the Goal when you are showing the subgoals for that goal.
2. Make navigation happen on first click.

**Describe alternatives you've considered**
The current implementation. It is easier to learn in the beginning, but then harder in thr long run. 
It is more acceptable to add an initial explanation (tour conponent) that you can tap the circle for the goal options.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
resolves #1924

### CODE REVIEW

**Review by tijlleenders — CHANGES_REQUESTED** [review]
Excellent!

Can you replace the pipe | with a comma ,?

Also please update the tests (or migrate and update them to Playwright if not much effort).

**Comment by vinaybadgujar102** [comment]
> Also please update the tests (or migrate and update them to Playwright if not much effort).

@tijlleenders, I think for smaller UI checks and functionalities, we can use a unit testing library. Playwright is an E2E testing library, and E2E tests are expensive to run and hard to maintain. So, for complex UI interactions (like the collaboration feature), we can use Playwright. For smaller UI and feature testing, we can rely on unit testing libraries like Jest or Vitest.

**Comment by tijlleenders** [comment]
Different tests for different purposes, I agree. 

Since the app is still very much in flux - any test is extra work that might get invalidated soon. So I only want to do end-to-end testing on major features.

For now I don't think investing in unit tests is worth it, unless you do it as part of your TDD flow.  

So for any test currently failing please explain what it is testing - so we can decide on a case-by-case basis what the appropriate level of the test is - and if we want it at this point in the project.

**Comment by tijlleenders** [comment]
@Tushar-4781 @vinaybadgujar102 Do you agree to invest in lower level test sparingly - as the expected rate of change is higher on the internals?

**Comment by vinaybadgujar102** [comment]
> So for any test currently failing please explain what it is testing - so we can decide on a case-by-case basis what the appropriate level of the test is - and if we want it at this point in the project.

We currently have 6 failing Cypress tests:

- **Goals Page Title:** Verifies that the page title is displayed correctly on the Goals Page.
- **Search Input Visibility:** Checks if the search input is shown when the search icon is clicked.
- **Settings Dropdown Open:** Ensures the settings dropdown opens when the settings icon is clicked.
- **Close Settings Dropdown:** Verifies that the settings dropdown closes when clicking outside of it.
- **Dark Mode Toggle:** Tests the functionality of toggling dark mode when the Switch Mode button is clicked.
- **Conditional Settings Display:** Ensures the Zinzen settings are only displayed on the MyJournal or My Time pages.

**Comment by tijlleenders** [comment]
@vinaybadgujar102 These are all high-level UI checks - for which I think it is appropriate to use a UI testing framework like Cypress/Playwright - don't you?  I have a hard time imagining how you would even do a UI check using a unit testing framework. For me unit testing is a relatively simple test on a specific function or module.  
No idea what the last test is for though. What is this 'settings display'?

**Comment by vinaybadgujar102** [comment]
@tijlleenders, I'm relatively new to testing, but I've learned that basic UI checks can be handled using a unit testing library like Jest. I think @Tushar-4781 might have some valuable insights to add on

>  No idea what the last test is for though. What is this 'settings display'?

The last test ensures that search button is only visible on Goals page and not others.

**Review by tijlleenders — DISMISSED** [review]
Awesome @vinaybadgujar102 !

@Tushar-4781 Code OK?

File: src/hooks/useGoalSummary.ts
Comment (Tushar-4781) [inline src/hooks/useGoalSummary.ts:1]: can we just use "?" that would save some LOC(s)

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]

File: src/hooks/useGoalSummary.ts
Comment (vinaybadgujar102) [reply src/hooks/useGoalSummary.ts:1]: Are you suggesting replacing the if statements?

Comment (Tushar-4781) [reply src/hooks/useGoalSummary.ts:1]: yup

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**